syntax = "proto2";

package transactions;

message Transaction {
  // ID of the node (tile, CGRA, CPU, etc.).
  optional int32 nodeID = 1;
  // Memory address. This will get translated to block address in the simulators.
  optional int32 memAddr = 2;
  // Transaction direction. 1 == RD, 0 == WR.
  optional int32 direction = 3;
  // Timestamp of the transaction.
  optional int32 tick = 4;
}

message Transaction_list {
  repeated Transaction transactions = 1;
}

